src/peer_digest.cc

```diff
@@ -699,7 +650,7 @@ peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const
     const char *reason = nullptr;  /* reason for completion */
     const char *no_bug = nullptr;  /* successful completion if set */
     const int pdcb_valid = cbdataReferenceValid(fetch->pd);
-    const int pcb_valid = cbdataReferenceValid(fetch->pd->peer);
+    const int pcb_valid = pdcb_valid && fetch->pd->peer.valid();
 
     /* test possible exiting conditions (the same for most steps!)
      * cases marked with '?!' should not happen */
```

Comment:
Please add `!pdcb_valid || ` as the first condition of the `if (!(pd = fetch->pd))` statement below. Since we are fixing `fetch->pd` dereference bugs in this method, let's fix the last(?) one of them.

Reply:
Done (70cb627).